### PR TITLE
migrate pg to new plugin system

### PR DESCRIPTION
**Pull request by khanayan123**
### What does this PR do?
migrates postgres to new plugin system

### Motivation
to convert plugins to new plugin system

### Plugin Checklist

- [x] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/dd-trace-js/blob/master/docs/API.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

File: packages/datadog-instrumentations/src/pg.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      if (error) {
+        errorCh.publish(error)
+      }
+      asyncEndCh.publish(undefined)
+    }
+
+    if (originalCallback) {
+      pgQuery.callback = AsyncResource.bind(function (err, res) {
+        finish(err)
+        // originalCallback(err, res)
```
Comment (bengl) [inline packages/datadog-instrumentations/src/pg.js:58]: Unnecessary comment.

File: packages/datadog-plugin-pg/src/t_index.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
+'use strict'
```
Comment (bengl) [inline packages/datadog-plugin-pg/src/t_index.js:1]: You probably intended to delete this file.

**Review by bengl — CHANGES_REQUESTED** [review]

File: packages/datadog-instrumentations/src/pg.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    if (!pgQuery) {
+      return retval
+    }
+    const statement = pgQuery.text
+
+    startCh.publish({ params: this.connectionParameters, statement })
+
+    let originalCallback = pgQuery.callback
+
+    const finish = function (error) {
```
Comment (rochdev) [inline packages/datadog-instrumentations/src/pg.js:48]: This should be bound explicitly.

Comment (rochdev) [reply packages/datadog-instrumentations/src/pg.js:48]: Just saw it's bound later, but let's bind it here instead to avoid multiple bindings.

**Review by rochdev — CHANGES_REQUESTED** [review]

File: packages/datadog-instrumentations/src/pg.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    const finish = AsyncResource.bind(function (error) {
+      if (error) {
+        errorCh.publish(error)
+      }
+      asyncEndCh.publish(undefined)
+    })
+
+    if (pgQuery.callback) {
+      const originalCallback = asyncResource.bind(pgQuery.callback)
+      pgQuery.callback = AsyncResource.bind(function (err, res) {
```
Comment (rochdev) [inline packages/datadog-instrumentations/src/pg.js:55]: This bind is no longer needed since both the callback and finish are bound above.

**Review by bengl — DISMISSED** [review]

**Review by rochdev — APPROVED** [review]

**Pull request merged**